### DISCUSSION TITLE
[Feature]: output.assetPrefix support to configure different values according to different types

### DISCUSSION THREAD

**Discussion by chenjinxinlove** [web-infra-dev/rsbuild#691] [General]
### What problem does this feature solve?

output.assetPrefix now supports configuring only one value, but sometimes the cnd address of js and the cnd address of an image are different. Different cdn prefixes need to be configured according to different types.

### What does the proposed API look like?

output:{
    assetPrefix: {
      js: 'cdn1',
      css: 'cdn1',
      image: 'cdn2'
 }
}

**Comment by chenjiahan** [comment]
We can not implement this feature yet, because the `output.publicPath` option of Rspack only accept a string value. see https://rspack.dev/config/output#outputpublicpath.

If Rspack supports `output.publicPath` to be a function, then Rsbuild can provide a more flexible `assetPrefix` config.

Can you submit an issue for Rspack to support `output.publicPath` function first? ❤️

https://github.com/web-infra-dev/rspack/issues
